app/src/test/java/com/saltedge/authenticator/features/main/MainActivityViewModelTest.kt

```diff
@@ -0,0 +1,640 @@
+ * under Section 7 of the GNU General Public License see THIRD_PARTY_NOTICES.md
+ */
+package com.saltedge.authenticator.features.main
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.test.core.app.ApplicationProvider
+import com.saltedge.authenticator.R
+import com.saltedge.authenticator.app.ConnectivityReceiverAbs
+import com.saltedge.authenticator.app.KEY_DEEP_LINK
+import com.saltedge.authenticator.app.QR_SCAN_REQUEST_CODE
+import com.saltedge.authenticator.features.menu.MenuItemData
+import com.saltedge.authenticator.models.ViewModelEvent
+import com.saltedge.authenticator.models.realm.RealmManagerAbs
+import com.saltedge.authenticator.sdk.constants.KEY_AUTHORIZATION_ID
+import com.saltedge.authenticator.sdk.constants.KEY_CONNECTION_ID
+import com.saltedge.authenticator.sdk.model.appLink.ActionAppLinkData
+import com.saltedge.authenticator.sdk.model.appLink.ConnectAppLinkData
+import com.saltedge.authenticator.sdk.model.authorization.AuthorizationIdentifier
+import org.hamcrest.CoreMatchers.*
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.BDDMockito.given
+import org.mockito.Mockito
+import org.mockito.Mockito.mock
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class MainActivityViewModelTest {
+
+    @Test
+    @Throws(Exception::class)
+    fun initTestCase1() {
+         //given
+        given(mockRealmManager.initialized).willReturn(true)
+
+        //when
+        val viewModel = createViewModel()
+
+        //then
+        Mockito.never()
+
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun initTestCase2() {
+        //given
+        given(mockRealmManager.initialized).willReturn(false)
+
+        //when
+        createViewModel()
+
+        //then
+        Mockito.verify(mockRealmManager).initRealm(context)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun onLifeCycleCreateTestCase1() {
+        /**
+         * given null savedInstanceState, null intent
+         */
+        val viewModel = createViewModel()
+        val savedInstanceState: Bundle? = null
+        val intent: Intent? = null
+        assertThat(viewModel.onShowAuthorizationsListEvent.value, `is`(nullValue()))
+        assertThat(viewModel.onShowAuthorizationDetailsEvent.value, `is`(nullValue()))
+
+        //when
+        viewModel.onLifeCycleCreate(savedInstanceState, intent)
+
+        //then onShowAuthorizationsListEvent is posted
+        assertThat(viewModel.onShowAuthorizationsListEvent.value, equalTo(ViewModelEvent(Unit)))
+        assertThat(viewModel.onShowAuthorizationDetailsEvent.value, `is`(nullValue()))
+        assertThat(viewModel.onShowConnectEvent.value, `is`(nullValue()))
```

Comment:
Mb smth like this : `assertNull(viewModel.onShowConnectEvent.value))` ?

Reply:
i'd like to unify all assertions. assertNull belongs to anower library, but is not forbiden to use it. on your wish